src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collections/TimeExpiringSet.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collections;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.util.Iterator;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+
+/**
+ * Stores a set of elements which are automatically removed from the Set after a given time period.
+ *
+ * <p>Subsequent calls to add with the same element refresh the expiry period for that element.
+ */
+public class TimeExpiringSet<E> implements Iterable<E> {
+  private Cache<E, E> cache;
+
+  /**
+   * Allocates a new TimeExpiringSet whose elements expire after the given time period
+   * @param duration The magnitude of the expiry duration
+   * @param unit The unit of the expiry duration
+   */
+  public TimeExpiringSet(long duration, TimeUnit unit) {
```

Comment:
Can we rename `duration` to a more appropriate variable name?  `expiryTimeout` perhaps ?

Reply:
Done